### PR TITLE
optimizer: fix optimizer hint identifier

### PR DESCRIPTION
**Pull request by AilinKid**
Signed-off-by: AilinKid <314806019@qq.com>

### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.6 (TiDB 6.6 versions)
- [x] v6.5 (TiDB 6.5 versions)
- [x] v6.4 (TiDB 6.4 versions)
- [x] v6.1 (TiDB 6.1 versions)
- [x] v5.4 (TiDB 5.4 versions)
- [x] v5.3 (TiDB 5.3 versions)
- [x] v5.2 (TiDB 5.2 versions)
- [x] v5.1 (TiDB 5.1 versions)
- [x] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by winoros — APPROVED** [review]

File: optimizer-hints.md [hunk 1]
```diff
@@ -7,6 +7,31 @@ aliases: ['/docs-cn/dev/optimizer-hints/','/docs-cn/dev/reference/performance/op
 
 TiDB 支持 Optimizer Hints 语法，它基于 MySQL 5.7 中介绍的类似 comment 的语法，例如 `/*+ HINT_NAME(t1, t2) */`。当 TiDB 优化器选择的不是最优查询计划时，建议使用 Optimizer Hints。
 
+> **注意：**
```
Comment (ran-huang) [inline optimizer-hints.md:10]: 这个内容放到 L39 （## 语法）这部分里是否更合适呢？

Comment (AilinKid) [reply optimizer-hints.md:10]: 可以，我想醒目一点，放 L39 （## 语法）开始部分可以吧

File: optimizer-hints.md [hunk 2]
```diff
@@ -13,6 +13,31 @@ TiDB 支持 Optimizer Hints 语法，它基于 MySQL 5.7 中介绍的类似 comm
 
 ## 语法
 
+> **注意：**
```
Comment (ran-huang) [inline optimizer-hints.md:16]:
```suggestion
> **注意：**
>
```

File: optimizer-hints.md [hunk 3]
```diff
@@ -13,6 +13,31 @@ TiDB 支持 Optimizer Hints 语法，它基于 MySQL 5.7 中介绍的类似 comm
...
+> 如果需要提示优化器使用的表不在 USE DATABASE 所指定的数据库内，需要显式指定数据库名。例如：
```
Comment (ran-huang) [inline optimizer-hints.md:17]:
```suggestion
> 如果需要提示优化器使用的表不在 `USE DATABASE` 所指定的数据库内，需要显式指定数据库名。例如：
```

File: optimizer-hints.md [hunk 4]
```diff
@@ -13,6 +13,31 @@ TiDB 支持 Optimizer Hints 语法，它基于 MySQL 5.7 中介绍的类似 comm
...
+>
+> tidb> SELECT /*+ HASH_JOIN(test2.t2, t) */ * FROM t, test2.t2;
+> Empty set (0.00 sec)
+> 
+> tidb> SELECT /*+ READ_FROM_STORAGE(TIFLASH[test1.t1,test2.t2]) */ t1.a FROM test1.t t1, test2.t t2 WHERE t1.a = t2.a;
+> Empty set (0.00 sec)
+> 
+> ```
+>
+> 后续示列演示部分，皆是同一个 database scope 范围内的表。
```
Comment (ran-huang) [inline optimizer-hints.md:39]:
```suggestion
> 本文档中后续示例演示部分，皆是同一个数据库范围内的表。如果你使用的表不在同一个数据库内，请参照指示显式指定数据库名。
```

**Review by ran-huang — APPROVED** [review]
Rest LGTM

**Pull request merged**